### PR TITLE
Separate Curse of Madness into two rituals: Brain curse, Curse of Twisted Reality 

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Separate Curse of Madness into two rituals:
* **Brain curse**: Everyone gets a special trauma for 20% chance, and magical trauma for 80% chance. Curable by Brain Recalibration.
* **Curse of Twisted Reality** : You write a message, and wait for an admin's approval. It works to similar to brainwash, but it will need to be creative to be approved. for example:
  * Captain is the lowest rank in Chain of Command.
  * Nanotrasen is a subcompany of Syndicate.
  * Hugging someone means I want to kill you and I will kill you.
  * Everyone is clown.
  * Card Duel is an actual valid way to kill a person. Harming a person directly without winning a card game is not a thing.
  
  It shouldn't work as a lame brainwash objective like `serve X, help them for their objectives.`, and that's why it has admin approval. ion laws will be good examples.
  This is also castable multiple times.

## Why It's Good For The Game
It's a ritual that nobody actually uses and a possible noob trap.
Renovating it into some form will make a use for engaging interesting RP moments

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

image

image

latejoin suffering

image

posibrain

image

round end

image

approval chat

</details>

## Changelog
:cl:
balance: curse of madness wizard ritual is now Brain Curse - 2 spell points. Curable by Brain Recalibration. Doesn't send a message of madness. 20% chance to get special trauma, 80% chance to get magical trauma.
add: new ritual - curse of twisted reality. It works as a mass brainwash, but it needs admin approval. 2 spell points. Late joiners will get the same suffering. Also, castable multiple times.
tweak: Curse of Madness wizard event now uses the curse of twisted reality wizard ritual instead of giving traumas to people randomly.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 1]
```diff
@@ -573,23 +573,74 @@
...
+						(<A HREF='?_src_=holder;[HrefToken(TRUE)];time_until=[world.time+wait_time];spellref=[REF(src)];ritual_reject=[REF(src)];spelluser=[REF(user)];spellbook=[REF(book)]'>REJECT</A>)</span>")
+	response_timer_id = addtimer(CALLBACK(src, PROC_REF(rejected), user, book), wait_time, TIMER_STOPPABLE)
+
+/datum/spellbook_entry/summon/curse_of_twisted_reality/proc/rejected(mob/living/carbon/human/user, obj/item/spellbook/book)
+	if(response_timer_id)
+		deltimer(response_timer_id)
+		response_timer_id = null
+	if(!active)
+		return
+	to_chat(user, "<span class='notice'>Wizard Federation rejected your ritual.</span>")
```
Comment (Absolucy) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	to_chat(user, "<span class='notice'>The Wizard Federation rejected your ritual.</span>")
```

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 2]
```diff
@@ -573,23 +573,74 @@
...
+	book.uses += cost
+	active = FALSE
+
+/datum/spellbook_entry/summon/curse_of_twisted_reality/proc/accepted(mob/living/carbon/human/user, message)
+	deltimer(response_timer_id)
+	response_timer_id = null
+	if(!active)
+		return
+	to_chat(user, "<span class='notice'>The twisted reality has been manipulated!</span>")
```
Comment (Absolucy) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	to_chat(user, "<span class='notice'>Reality has been twisted!</span>")
```

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 3]
```diff
@@ -573,23 +573,74 @@
...
+/datum/spellbook_entry/summon/curse_of_twisted_reality/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
+	SSblackbox.record_feedback("tally", "wizard_spell_learned", 1, name)
+	active = TRUE
+	var/message = stripped_input(user, "Declare the new fact to twist the reality of everyone.", "Twisting the reality")
+	if(!message)
+		to_chat(user, "<span class='notice'>You cancel the ritual.</span>")
+		active = FALSE
+		return FALSE
+
+	to_chat(user, "<span class='notice'>Wizard Federation will evaluate your declaration: [message]. \
```
Comment (Absolucy) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	to_chat(user, "<span class='notice'>The Wizard Federation will evaluate your declaration: [message]. \
```

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 4]
```diff
@@ -573,23 +573,74 @@
...
+			Once a ritual is rejected, your points are refuneded. This ritual is not included in Wild Magic Manipulation."
+	cost = 2
+	ritual_invocation = "ALADAL DESINARI ODORI'IN PORES ANTE EGUL'OVOR'E MODU ENCANTOLITE"
+	var/wait_time = 600
+	var/response_timer_id = null
+
+/datum/spellbook_entry/summon/curse_of_twisted_reality/Buy(mob/living/carbon/human/user, obj/item/spellbook/book)
+	SSblackbox.record_feedback("tally", "wizard_spell_learned", 1, name)
+	active = TRUE
+	var/message = stripped_input(user, "Declare the new fact to twist the reality of everyone.", "Twisting the reality")
```
Comment (Absolucy) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	var/message = tgui_input_text(user, "Declare the new fact to twist the reality of everyone.", "Twisting the reality")
```
note: tgui_input_text automatically encodes/trims/etc.

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 5]
```diff
@@ -573,23 +573,74 @@
...
 
+/datum/spellbook_entry/summon/curse_of_twisted_reality
+	name = "Curse of Twisted Reality"
+	desc = "Curse everyone's reality as what they know is not what they know in fact, even including yourself. \
+			<br><b>Warning:</b> Wizard Federation Administrative Magus will evaluate your declaration, \
+			and decide to approve it or not. If there's no available person, it will be automatically rejected. \
+			Once a ritual is rejected, your points are refuneded. This ritual is not included in Wild Magic Manipulation."
+	cost = 2
+	ritual_invocation = "ALADAL DESINARI ODORI'IN PORES ANTE EGUL'OVOR'E MODU ENCANTOLITE"
+	var/wait_time = 600
```
Comment (Absolucy) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	var/wait_time = 1 MINUTES
```

File: code/datums/mind.dm [hunk 6]
```diff
@@ -401,6 +401,12 @@
 	var/output = "<B>[current.real_name]'s Memories:</B><br>"
 	output += memory
 
+	if(length(GLOB.curse_of_twisted_reality_messages))
+		output += "<br>Your mind is imprinted with the fact of the twisted reality:<BR>"
```
Comment (Absolucy) [inline code/datums/mind.dm]:
```suggestion
		output += "<br>Your mind is imprinted with the fact[length(GLOB.curse_of_twisted_reality_messages) > 1 ? "s" : ""] of the twisted reality:<BR>"
```

**Review by Absolucy — CHANGES_REQUESTED** [review]

**Comment by EvilDragonfiend** [comment]
Thanks for the grammar corrections lol

**Comment by Absolucy** [comment]
Glad to help. This seems like a really fun idea, especially since admin approval makes this hard to abuse. I hope it gets through (and I hope I manage to roll wizard one day)

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Requiring admin approval for something is indicative of poor design. If you are going to require admin approval it shouldn't be something like "must be creative", rather "must not be a rule break".

Also I don't really understand how this is good for the game. The reasoning in the PR body is that nobody buys it, which is untrue, but even if that was the case wouldn't splitting it into 2 spells make ot objectively worse and less likely to be purchased?

**Comment by EvilDragonfiend** [comment]
Also, the required message manner to the ritual is ion laws style. It shouldn't be just typical brainwash message, but people will not understand how they should write a thing.

**Comment by PowerfulBacon** [comment]
The current curse of madness isn't brainwashing, people don't get brainwashed by it. It is a message inside their heads which drives them mad and applies the magical traumas to everyone.

**Comment by JixS4v** [comment]
Maybe implement a suggestion system that lets the player select some pre defined brainwashes that are similar to the expected player-provided curses?

**Comment by PowerfulBacon** [comment]
I think giving the wizard the ability to just brainwash everyone is a bad idea, the current curse is a message and not a brainwash. Especially since this requires strict admin moderation on top of it to function.

**Pull request closed without merging**